operator/controllers/keptnevaluation/controller.go

```diff
@@ -244,19 +249,96 @@ func (r *KeptnEvaluationReconciler) fetchDefinitionAndProvider(ctx context.Conte
 func (r *KeptnEvaluationReconciler) queryEvaluation(objective klcv1alpha1.Objective, provider klcv1alpha1.KeptnEvaluationProvider) *klcv1alpha1.EvaluationStatusItem {
 	query := &klcv1alpha1.EvaluationStatusItem{
 		Value:  "",
-		Status: common.StateSucceeded, //setting status per default to failed
+		Status: common.StateFailed, //setting status per default to failed
 	}
 
-	//TODO query provider like prometheus service does, save result in value THIS SHALL BE SOLVED IN TICKET #163
-	// it will be something hardcoded like
-	// import apiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
-	// if provider ==prometheus   {  result, w, err := apiv1.PrometheusAPI.Query(context.Background(), query, endUnix)	if err != nil {		return 0, fmt.Errorf("unable to query prometheus api: %w", err)}}
-	//TODO check value with evaluation target and update status in query
-	// result, w, err := prometheus.API().Query(context.Background(), query, time.Now())
+	queryTime := time.Now()
+	r.Log.Info("Running query: /api/v1/query?query=" + objective.Query + "&time=" + queryTime.String())
+
+	client, err := promapi.NewClient(promapi.Config{Address: provider.Spec.TargetServer, Client: &http.Client{}})
+	api := prometheus.NewAPI(client)
+	result, w, err := api.Query(
+		context.Background(),
+		objective.Query,
+		queryTime,
+		[]prometheus.Option{}...,
+	)
+
+	if err != nil {
+		query.Value = err.Error()
+		return query
+	}
+
+	if len(w) != 0 {
+		r.Log.Info("Prometheus API returned warnings: " + w[0])
+	}
+
+	// check if we can cast the result to a vector, it might be another data struct which we can't process
+	resultVector, ok := result.(model.Vector)
+	if !ok {
+		return query
```

Comment:
should we return an error here to give better feedback about a query not returning the data in the format we require?